### PR TITLE
[SPARK-47671][Core] Enable structured logging in log4j2.properties.template and update docs

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

- Rename the current log4j2.properties.template as log4j2.properties.pattern-layout-template
- Enable structured logging in log4j2.properties.template
- Update `configuration.md` on how to configure logging

### Why are the changes needed?

Providing a structured logging template and document how to configure loggings in Spark 4.0.0

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manual test

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @dtenedor @panbingkun as well.

File: docs/configuration.md [hunk 1]
```diff
@@ -3670,14 +3670,17 @@ Note: When running Spark on YARN in `cluster` mode, environment variables need t
...
 
-By default, Spark adds 1 record to the MDC (Mapped Diagnostic Context): `mdc.taskName`, which shows something
-like `task 1.0 in stage 0.0`. You can add `%X{mdc.taskName}` to your patternLayout in
-order to print it in the logs.
+## Structured Logging
+Starting from version 4.0.0, Spark has adopted the [JSON Template Layout](https://logging.apache.org/log4j/2.x/manual/json-template-layout.html) for logging, which outputs logs in JSON format. This format facilitates querying logs using Spark SQL with the JSON data source. Additionally, the logs include all Mapped Diagnostic Context (MDC) information for search and debugging purposes.
+
+To implement structured logging, start with the `log4j2.properties.template` file.
+
+## Plain Text Logging
```
Comment (dongjoon-hyun) [inline docs/configuration.md:3680]: Thank you for providing this section together.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by gengliangwang** [comment]
@dongjoon-hyun I double checked and removed some comments. PTAL, thanks!

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you. Merged to master, @gengliangwang !

**Pull request closed without merging**